core/src/main/scala/kafka/tools/GetOffsetShell.scala

```diff
@@ -132,23 +150,104 @@ object GetOffsetShell {
         }
     }
 
-    partitionOffsets.toArray.sortBy { case (tp, _) => tp.partition }.foreach { case (tp, offset) =>
-      println(s"$topic:${tp.partition}:${Option(offset).getOrElse("")}")
+    partitionOffsets.toSeq.sortWith((tp1, tp2) => {
```

Comment:
I just learnt that we can actually compare TopicPartition with the following comparator in Scala:

```
  def compareTopicPartitions(a: TopicPartition, b: TopicPartition): Boolean = {
    (a.topic(), a.partition()) < (b.topic(), b.partition())
  }
```

and do `partitionOffsets.toSeq.sortWith(compareTopicPartitions).foreach { ...`. This makes the code a bit more readable.

Reply:
Thanks, refactored it